### PR TITLE
Simplify object client methods

### PR DESCRIPTION
**Pull request by fyrchik**
It is nice to have all control-flow in a single file (ideally, on a single screen). In this PR we get rid of using multiple callbacks in object* methods. The result uses less code and is simpler to follow IMO.

### CODE REVIEW

File: client/object_search.go [hunk 1]
```diff
@@ -243,75 +233,37 @@ func (c *Client) ObjectSearchInit(ctx context.Context, prm PrmObjectSearch) (*Ob
...
-		stream, err = rpcapi.SearchObjects(&c.c, &req, client.WithContext(ctx))
-		if err != nil {
-			return fmt.Errorf("open stream: %w", err)
-		}
-
-		return nil
-	}
-	r.ctxCall.rResp = func() error {
-		return stream.Read(&resp)
+	r.stream, err = rpcapi.SearchObjects(&c.c, &req, client.WithContext(ctx))
```
Comment (carpawell) [inline client/object_search.go:262]: just wanna highlight that change one more time: we now open stream in the init call, not in a reader. however it look more like a fix cause all our stream RPC has `The call only opens the transmission channel, explicit recording is done using the *.`

could lead to some unexpected changes in the clients (mb SN)

Comment (fyrchik) [reply client/object_search.go:262]: Yes, I have researched a bit -- pool and neofs-cli use `Init` immediately followed by `WriteHeader`, so everything should be ok.

Comment (fyrchik) [reply client/object_search.go:258]: Actually, `opens the transmission channel` message shouldn't even be changed -- this is what we do here, open the stream for further processing.

Comment (carpawell) [reply client/object_search.go:258]: > shouldn't even be changed

yea, that is why i called it more like a fix not a refactor in that case

**Review by carpawell — APPROVED** [review]

**Review by alexvanin — APPROVED** [review]
LGTM, DNT

**Comment by alexvanin** [comment]
Closes https://github.com/nspcc-dev/neofs-sdk-go/issues/324?

**Comment by KirillovDenis** [comment]
> Closes #324?

It seems so

**Pull request merged**